sdk/core/azure-core/src/main/java/com/azure/core/util/implementation/ByteArrayContent.java

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util.implementation;
+
+import com.azure.core.util.serializer.ObjectSerializer;
+import com.azure.core.util.serializer.TypeReference;
+import reactor.core.publisher.Flux;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.Objects;
+
+/**
+ * A {@link BinaryDataContent} implementation which is backed by a {@code byte[]}.
+ */
+public final class ByteArrayContent extends BinaryDataContent {
+    private final byte[] content;
+
+    /**
+     * Creates a new instance of {@link ByteArrayContent}.
+     *
+     * @param content The byte array content.
+     * @throws NullPointerException if {@code content} is null.
+     */
+    public ByteArrayContent(byte[] content) {
+        this.content = Objects.requireNonNull(content, "'content' cannot be null");
```

Comment:
Should the non-null check in `BinaryData` be removed as it is being performed here?